### PR TITLE
Broker Merger

### PR DESCRIPTION
**Pull request by kamilczaja**
_What issues does this PR close?_

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [x] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/web/auth/ConfigApiKeyFilter.kt
Comment (kamilczaja) [inline authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/web/auth/ConfigApiKeyFilter.kt:1]: My impression is that this file is unused.
A comment in `ConfigResourceImpl.kt` is saying that this is handling the auth for config endpoints via API Key but I wasn't able to find a reference in code. Could you take a look at it @richardtreier ?

File: old-tests/integration/pages/organizationmanagement/OrganizationInfoApiServiceTest.kt [hunk 1]
```diff
@@ -90,7 +117,7 @@ class OrganizationInfoApiServiceTest {
         assertThat(organization1.registrationStatus).isEqualTo(testRegistrationStatus)
     }
 
-    @Test
+    // TODO Fix test
```
Comment (richardtreier) [inline old-tests/integration/pages/organizationmanagement/OrganizationInfoApiServiceTest.kt:120]: todo

File: old-tests/integration/pages/organizationmanagement/OrganizationInfoApiServiceTest.kt [hunk 2]
```diff
@@ -147,7 +174,7 @@ class OrganizationInfoApiServiceTest {
         verify(keycloakService).getUserRoles(eq(userId))
     }
 
-    @Test
+    // TODO Fix test
```
Comment (richardtreier) [inline old-tests/integration/pages/organizationmanagement/OrganizationInfoApiServiceTest.kt:177]: todo

**Review by richardtreier — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,25 +11,53 @@ please see [changelog_updates.md](docs/dev/changelog_updates.md).
...
+  It should be noted that the Catalog can only display and save crawled data offers but cannot actually fetch the data.
+  For this functionality, an EDC Connector with the "Broker Crawler" extension is required. For more information, see the [Deployment Migration Notes](#deployment-migration-notes).
+
 #### Minor
 
+- Due to the integration of the Data Catalog into the portal, following changes have been made:
+  - Removed the link to "My data offers" from the sidebar. Due to the integration of the Data Catalog, this feature is not supported for now.
+    Users may access a view of their data offers by navigating to the Data Catalog and choosing the appropriate filter.
+  - Removed the Data Catalog online status from the dashboard
+  - Removed the Data Catalog status data from the system stability report
```
[the remark this answers is not in the document]
Comment (richardtreier) [reply CHANGELOG.md:22]: moved to https://github.com/sovity/PMO-Software/issues/1261

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,25 +11,53 @@ please see [changelog_updates.md](docs/dev/changelog_updates.md).
...
 #### Major
 
+- The Data Catalog (formerly Broker Server) is now integrated into the Authority Portal and can be accessed via the sidebar by authenticated users. 
+  It should be noted that the Catalog can only display and save crawled data offers but cannot actually fetch the data.
+  For this functionality, an EDC Connector with the "Broker Crawler" extension is required. For more information, see the [Deployment Migration Notes](#deployment-migration-notes).
+
 #### Minor
 
+- Due to the integration of the Data Catalog into the portal, following changes have been made:
+  - Removed the link to "My data offers" from the sidebar. Due to the integration of the Data Catalog, this feature is not supported for now.
```
[the remark this answers is not in the document]
Comment (richardtreier) [reply CHANGELOG.md:19]: moved to https://github.com/sovity/PMO-Software/issues/1261

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,25 +11,53 @@ please see [changelog_updates.md](docs/dev/changelog_updates.md).
...
+      authority-portal.deployment.environments.{environmentId}.broker.catalog-page-page-size: 10
+      # Default dataspace name
+      authority-portal.deployment.environments.{environmentId}.broker.dataspace-names.default: MDS
+      ```
+  - Following environment variables have been removed and **can be removed from the configuration**
+    - ```yaml
+      authority-portal.deployment.environments.{environmentId}.broker.url
+      authority-portal.deployment.environments.{environmentId}.broker.admin-api-key
+      authority-portal.deployment.environments.{environmentId}.broker.api-key
+      authority-portal.deployment.environments.{environmentId}.broker.kuma-name
```
[the remark this answers is not in the document]
Comment (richardtreier) [reply CHANGELOG.md]: moved to https://github.com/sovity/PMO-Software/issues/1261

File: old-tests/integration/pages/organizationmanagement/OrganizationInfoApiServiceTest.kt (hunk 1 above)
Comment (richardtreier) [reply old-tests/integration/pages/organizationmanagement/OrganizationInfoApiServiceTest.kt:120]: https://github.com/sovity/PMO-Software/issues/1261

File: authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/web/auth/ConfigApiKeyFilter.kt
Comment (richardtreier) [reply authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/web/auth/ConfigApiKeyFilter.kt:1]: i think it's in use

**Review by richardtreier — APPROVED** [review]
all further stuff will be done as separate PRs

**Pull request merged**